test/functional/eval/capture_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')
+local eq = helpers.eq
+local eval = helpers.eval
+local clear = helpers.clear
+local execute = helpers.execute
+local source = helpers.source
+
+describe('capture()', function()
+  before_each(clear)
+
+  it('returns the same result with :redir', function()
+    execute('redir => g:foo | messages | redir END')
+    eq(eval('g:foo'), eval('capture(":messages")'))
+  end)
+
+  it('returns the output of the commands if the argument is List', function()
+    eq("foobar", eval([[capture(['echon "foo"', 'echon "bar"'])]]))
+  end)
+
+  it('supports the nested redirection', function()
+    source([[
+    function! g:Foo()
+      let a = ''
+      redir => a
+      silent echon "foo"
+      redir END
+      return a
+    endfunction
+    function! g:Bar()
+      let a = ''
+      redir => a
+      call g:Foo()
+      redir END
+      return a
+    endfunction
+    ]])
+    eq('foo', eval('capture("call g:Bar()")'))
+  end)
```

Comment:
`capture('echo "\<C-a>"')` to check what exactly it captures (literal `^A` or byte 0x01).